charmhelpers/contrib/charmsupport/nrpe.py

```diff
@@ -260,11 +261,15 @@ def __init__(self, hostname=None, primary=True):
         relation = relation_ids('nrpe-external-master')
         if relation:
             log("Setting charm primary status {}".format(primary))
-            for rid in relation_ids('nrpe-external-master'):
+            for rid in relation:
                 relation_set(relation_id=rid, relation_settings={'primary': self.primary})
+        self.remove_check_queue = set()
 
     def add_check(self, *args, **kwargs):
         self.checks.append(Check(*args, **kwargs))
+        shortname = kwargs['shortname']
+        if shortname in self.remove_check_queue.copy():
+            del self.remove_check_queue[shortname]
```

Comment:
Won't this `TypeError` if it is a set?

A better way to do this is:

```python
try:
    self.remove_check_queue.remove(shortname)
except KeyError:
    pass
```

Reply:
rewrote that part following your suggestion.